test/unittest/dds/participant/ParticipantTests.cpp

```diff
@@ -63,10 +63,10 @@ TEST(ParticipantTests, DomainParticipantPSMListenerMethods)
 
     ASSERT_EQ(participant.listener(), nullptr);
 
-    ::dds::domain::DomainParticipantListener* listener;
-    participant.listener(listener, ::dds::core::status::StatusMask::all());
+    ::dds::domain::DomainParticipantListener listener = ::dds::domain::DomainParticipantListener();
```

Comment:
```suggestion
    ::dds::domain::DomainParticipantListener listener;
```
No need to construct-and-assign.